src/dbus.c

```diff
@@ -183,24 +332,23 @@ int main(int argc, char *argv[]) {
         goto finish;
     }
 
-    for (;;) {
-        r = sd_bus_process(bus, NULL);
-        if (r < 0) {
-            fprintf(stderr, "Failed to process bus: %s\n", strerror(-r));
-            goto finish;
-        }
-        if (r > 0)
-            continue;
-
-        /* Wait for the next request to process */
-        r = sd_bus_wait(bus, (uint64_t) -1);
-        if (r < 0) {
-            fprintf(stderr, "Failed to wait on bus: %s\n", strerror(-r));
-            goto finish;
-        }
+    r = sd_bus_attach_event(bus, event, SD_EVENT_PRIORITY_NORMAL);
+    if (r < 0) {
+        fprintf(stderr, "Failed to attach event loop: %s\n", strerror(-r));
+        goto finish;
     }
 
+    /* Mask the SIGCHLD signal, so we can listen to it via mainloop */
+    sigemptyset(&mask);
+    sigaddset(&mask, SIGCHLD);
+    sigprocmask(SIG_BLOCK, &mask, NULL);
+
+    /* Start the event loop, wait for requests */
+    r = sd_event_loop(event);
+    if (r < 0)
+        fprintf(stderr, "Failed mainloop: %s\n", strerror(-r));
 finish:
+    g_free(data);
```

Comment:
Do we also need/should do a `sd_event_unref(event);` here? Or maybe it's not really necessary?

Reply:
Absolutely! This is the proper way to clean it up. The application terminates afterwards, so it doesn't make that big of a difference, but still we should have it properly cleaned up in the code.

Fixed.